client/src/nv_ingest_client/client/job_manager.py

```diff
@@ -0,0 +1,348 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+
+import inspect
+from concurrent.futures import Future
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from nv_ingest_client.client.client import NvIngestClient
+from nv_ingest_client.primitives import BatchJobSpec
+from nv_ingest_client.primitives.jobs import JobStateEnum
+from nv_ingest_client.primitives.tasks import DedupTask
+from nv_ingest_client.primitives.tasks import EmbedTask
+from nv_ingest_client.primitives.tasks import ExtractTask
+from nv_ingest_client.primitives.tasks import FilterTask
+from nv_ingest_client.primitives.tasks import SplitTask
+from nv_ingest_client.primitives.tasks import StoreTask
+from nv_ingest_client.primitives.tasks import VdbUploadTask
+
+DEFAULT_JOB_QUEUE_ID = "morpheus_task_queue"
+
+
+class NvIngestJobManager:
+    """
+    NvIngestJobManager provides an interface for building, managing, and running data ingestion jobs
+    through NvIngestClient, allowing for chainable task additions and job state tracking.
+
+    Parameters
+    ----------
+    documents : List[str]
+        List of document paths to be processed.
+    client : Optional[NvIngestClient], optional
+        An instance of NvIngestClient. If not provided, a client is created.
+    job_queue_id : str, optional
+        The ID of the job queue for job submission, default is "morpheus_task_queue".
+    """
+
+    def __init__(
+        self,
+        documents: List[str],
+        client: Optional[NvIngestClient] = None,
+        job_queue_id: str = DEFAULT_JOB_QUEUE_ID,
+    ):
+        self._documents = documents
```

Comment:
I think we're going to want to pull in a library like fsspec, so we can handle remote paths, like s3://...., etc...